subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/InputPropertiesSerializer.java

```diff
@@ -28,7 +33,7 @@
     private final MapSerializer<String, Object> serializer;
 
     InputPropertiesSerializer(ClassLoader classloader) {
-        this.serializer = new MapSerializer<String, Object>(BaseSerializerFactory.STRING_SERIALIZER, new DefaultSerializer<Object>(classloader));
+        this.serializer = new MapSerializer<String, Object>(BaseSerializerFactory.STRING_SERIALIZER, new WellKnownTypesSerializer<Object>(classloader));
```

Comment:
This is all changing as part of the PR to better handle input properties (https://github.com/gradle/gradle/pull/962), so this change here will not be required.